scripts/agent-trader/agent_trader.py

```
@@ -296,46 +410,97 @@ def _latest_lessons_section() -> str:
+        if in_ul:
+            out.append("</ul>")
+            in_ul = False
+
     for raw in md.splitlines():
-        l = raw.rstrip()
-        if not l.strip():
-            if in_ul:
-                out.append("</ul>"); in_ul = False
+        line = raw.rstrip()
+        if not line.strip():
+            flush(); close_ul()
```

Ruff (0.16.1)

[error] Multiple statements on one line (semicolon)

(E702)